examples/wallet/test/components/SendTransaction.spec.js

```diff
@@ -0,0 +1,434 @@
+// @flow
+import { stub } from 'sinon';
+import { flatten } from 'lodash';
+import config from 'config';
+import React from 'react';
+import Button from 'react-bootstrap/Button';
+import Form from 'react-bootstrap/Form';
+import Enzyme, { shallow } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import renderer from 'react-test-renderer';
+import type { NodeSettings } from '../../app/reducers/types';
+import SendTransaction from '../../app/components/SendTransaction';
+
+Enzyme.configure({ adapter: new Adapter() });
+const defaultNodeSettings = {
+  block0Hash: 'notimportant',
+  fees: { constant: 10, certificate: 10, coefficient: 10 }
+};
+const mockedAddress = 'notimportant';
+const getAddressInput = form =>
+  form.find(Form.Control).find({ name: 'recipient' });
+const getAmountInput = form => form.find(Form.Control).find({ name: 'amount' });
+const getAddressFeedback = form =>
+  form
+    .find(Form.Control.Feedback)
+    .find({ type: 'invalid' })
+    .at(0);
+const getAmountNegativeFeedback = form =>
+  form
+    .find(Form.Control.Feedback)
+    .find({ type: 'invalid' })
+    .at(1);
+const getAmountPositiveFeedback = form =>
+  form.find(Form.Control.Feedback).find({ type: 'valid' });
+const getSubmitButton = form => form.find(Button);
+
+function setup(balance: number, nodeSettings: NodeSettings) {
```

Comment:
There is an error running test with "yarn test"
```
    console.log app/components/SendTransaction.js:34
      success!
    console.error app/components/SendTransaction.js:37
      couldnt send funds:  Error: Error
          at Object.rejects (/home/atixlabs/src/iohk/js-chain-libs/examples/wallet/node_modules/sinon/lib/sinon/defaul
t-behaviors.js:193:22)
          at Object.proto.<computed> [as rejects] (/home/atixlabs/src/iohk/js-chain-libs/examples/wallet/node_modules/sinon/lib/sinon/behavior.js:238:12)
          at Function.rejects (/home/atixlabs/src/iohk/js-chain-libs/examples/wallet/node_modules/sinon/lib/sinon/behavior.js:231:46)
          at Object.<anonymous> (/home/atixlabs/src/iohk/js-chain-libs/examples/wallet/test/components/SendTransaction.spec.js:118:19)
          at Object.asyncJestLifecycle (/home/atixlabs/src/iohk/js-chain-libs/examples/wallet/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:53:37)
          at /home/atixlabs/src/iohk/js-chain-libs/examples/wallet/node_modules/jest-jasmine2/build/queueRunner.js:43:12
          at new Promise (<anonymous>)
          at mapper (/home/atixlabs/src/iohk/js-chain-libs/examples/wallet/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
          at /home/atixlabs/src/iohk/js-chain-libs/examples/wallet/node_modules/jest-jasmine2/build/queueRunner.js:73:41
```

Reply:
Yes, that error is caused by a test that's supposed to fail (it tests the form behaviour when the transaction can't be generated/broadcasted for whatever reason).

The only alternative that I can think of is to have a logging module that maps to console calls in the real environment but does nothing (or perhaps can be stubbed?) when running the tests. Do you have any further ideas? Should I implement that?

Reply:
Yes, I think you should.